hikari/internal/typing_extensions.py

```diff
@@ -55,62 +51,79 @@ def __new__(
         global _Protocol
 
         if _Protocol is NotImplemented:
-            if cls_name != "Protocol":
-                raise TypeError("First instance of _FastProtocol must be Protocol")
+            if cls_name != "FastProtocolChecking":
```

Comment:
can't you just check against the class itself so it doesn't have collision risks?